packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -260,6 +262,60 @@ class PrecacheController {
     const urlObject = new URL(url, location.href);
     return this._urlsToCacheKeys.get(urlObject.href);
   }
+
+  /**
+   * Returns a function that looks up `url` in the precache (taking into
+   * account revision information), and returns the corresponding `Response`.
+   * 
+   * If for an unexpected reason there is a cache miss when looking up `url`,
+   * this will fall back to retrieving the `Response` via `fetch()`.
+   *
+   * @param {string} url The precached URL which will be used to lookup the
+   * `Response`.
+   * @return {workbox.routing.Route~handlerCallback}
+   */
+  createHandlerForURL(url: string): RouteHandlerCallback {
+    if (process.env.NODE_ENV !== 'production') {
+      assert!.isType(url, 'string', {
+        moduleName: 'workbox-precaching',
+        funcName: 'createHandlerForURL',
+        paramName: 'url',
+      });
+    }
+  
+    const cacheKey = this.getCacheKeyForURL(url);
+    if (!cacheKey) {
+      throw new WorkboxError('non-precached-url', {url});
+    }
+  
+    return async () => {
+      try {
+        const cache = await caches.open(this._cacheName);
+        const response = await cache.match(cacheKey);
+  
+        if (response) {
+          return response;
+        }
+  
+        // This shouldn't normally happen, but there are edge cases:
+        // https://github.com/GoogleChrome/workbox/issues/1441
+        throw new Error(`The cache ${this._cacheName} did not have an entry ` +
+            `for ${cacheKey}.`);
+      } catch (error) {
+        // If there's either a cache miss, or the caches.match() call threw
+        // an exception, then attempt to fulfill the navigation request with
+        // a response from the network rather than leaving the user with a
+        // failed navigation.
+        if (process.env.NODE_ENV !== 'production') {
+          logger.debug(`Unable to respond to navigation request with ` +
+              `cached response. Falling back to network.`, error);
+        }
+  
+        // This might still fail if the browser is offline...
+        return fetch(cacheKey);
```

Comment:
Do you want to fetch the `cacheKey` or the `url` in the fallback case?

Reply:
`cacheKey` makes the most sense, I think. Whenever `workbox-precaching` makes a network request, it's for the cache key (which includes the `__WB_REVISION` query parameter, as needed). Using the `cacheKey` here is consistent.

Reply:
Interesting. Doesn't that potentially bypass the HTTP cache in a way that may be unnecessary (or is that the point)?

Reply:
It's more likely to _hit_ the HTTP cache, since `/something.html?__WB_REVISION=1234` will likely have been requested by the service worker previously (in order to populate the precache), while `/something.html` hasn't necessarily been requested.

Getting to this point in the code is, by definition, a failure scenario, so I'm not sure what the expectation is for a developer. We're just making a best-case effort to (temporarily) return something when this happens.

Reply:
I see, my question was based on the assumption that this was different from the initial precache fetch, but it sounds like my assumption was incorrect. As long as the same URL is used in both cases, caching shouldn't be an issue.

I will say that I'm surprised the cache key is used in the initial precaching fetch (hadn't noticed that before because the `_addURLToCache()` method always references the `url` variable, which is not the same as the `url` variable used in `addToCacheList()`).

Anyway, if this is what's been happening for a while and no one has complained, I'm sure it's fine. My concern would have been that adding a query param to the URL that the developer may not be expecting might have unintended side effects -- depending on how their server is set up.

Reply:
Right, I think there's a decent amount of refactoring and renaming that could be done to the `workbox-precaching` codebase that would clarify some of those points.

You could visit a site that uses `workbox-precaching` in an Incognito window to confirm that `__WB_REVISION=...` is included in the outgoing request URLs.